pkgs/applications/version-management/gitlab/gitaly/default.nix

```diff
@@ -19,14 +19,14 @@ let
       };
   };
 in buildGoPackage rec {
```

Comment:
I noticed they have a go.mod now. can we ditch the 300+ line deps.nix's ? https://gitlab.com/gitlab-org/gitaly/-/blob/master/go.mod

Reply:
I think people have different opinions on which Go tooling should be used. I personally don't care so I changed it to buildGoModule and removed the deps.nix.

Reply:
Oh nevermind I forgot about other Go components of GitLab, I'll update this PR to remove all deps.nix files from GitLab components.